plugins/in_http/http_prot.c

```diff
@@ -1121,17 +1318,36 @@ static int process_pack_ng(struct flb_http *ctx, flb_sds_t tag, char *buf, size_
                          result.data.type);
 
             msgpack_unpacked_destroy(&result);
+            if (remote_addr) {
+                flb_sds_destroy(remote_addr);
+            }
 
             return -1;
         }
     }
 
     msgpack_unpacked_destroy(&result);
+    if (appended_initialized) {
+        msgpack_unpacked_destroy(&appended_result);
+        msgpack_sbuffer_destroy(&appended_sbuf);
+    }
+    if (remote_addr) {
+        flb_sds_destroy(remote_addr);
+    }
+
     return 0;
 
 log_event_error:
     flb_plg_error(ctx->ins, "Error encoding record : %d", ret);
     msgpack_unpacked_destroy(&result);
+    if (appended_initialized) {
+        msgpack_unpacked_destroy(&appended_result);
+        msgpack_sbuffer_destroy(&appended_sbuf);
+    }
+    if (remote_addr) {
```

Comment:
Non NULL checking would be better to use here:

```suggestion
    if (remote_addr != NULL) {
```